src/deepseek.rs

```diff
@@ -71,6 +72,81 @@ impl TryFrom<f32> for FrequencyPenalty {
     }
 }
 
+// Does this make sense or do we want to handle serialization somewhere else?
+#[derive(Debug)]
+pub enum Message {
+    System {
+        content: String,
+        name: Option<String>,
+    },
+    User {
+        content: String,
+        name: Option<String>,
+    },
+    Assistant {
+        content: Option<String>,
+        name: Option<String>,
+        prefix: Option<bool>,
+        reasoning_content: Option<String>,
+    },
+    Tool {
+        content: String,
+        tool_call_id: String,
+    },
+}
+
+impl Serialize for Message {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut map = serializer.serialize_map(None)?;
+        match self {
+            Message::System { content, name } => {
+                map.serialize_entry("role", "system")?;
+                map.serialize_entry("content", content)?;
+                if let Some(name) = name {
+                    map.serialize_entry("name", name)?;
+                }
+            }
+            Message::User { content, name } => {
+                map.serialize_entry("role", "user")?;
+                map.serialize_entry("content", content)?;
+                if let Some(name) = name {
+                    map.serialize_entry("name", name)?;
+                }
+            }
+            Message::Assistant {
+                content,
+                name,
+                prefix,
+                reasoning_content,
+            } => {
+                map.serialize_entry("role", "assistant")?;
+                map.serialize_entry("content", content)?;
+                if let Some(name) = name {
+                    map.serialize_entry("name", name)?;
+                }
+                if let Some(prefix) = prefix {
+                    map.serialize_entry("prefix", prefix)?;
+                }
```

Comment:
This whole `impl` smells. To get a feel of what should be done, start with `impl Message` and `fn role(&self) -> &'static str`. We should aim to be able to `#[derive(Serialize)]`